### PR TITLE
refactor(common) migrate from javacript to typescript

### PR DESCRIPTION
**Pull request by OmerDital**

### CODE REVIEW

File: client/App/components/Tabs/FirstTab/index.tsx [hunk 1]
```diff
@@ -33,6 +34,13 @@ export const FirstTab = () => {
     },
   });
 
+  const calculate = (event: KeyboardEvent) => {
```
Comment (benben12341) [inline client/App/components/Tabs/FirstTab/index.tsx:37]: destructure key from event

File: client/App/utilities/axios/generate-api.ts [hunk 2]
```diff
@@ -34,7 +35,7 @@ export const generateApi = ({ instance }) => {
     }
   };
 
-  const put = async ({ url, body }) => {
+  const put = async (url: string, body: any) => {
```
Comment (benben12341) [inline client/App/utilities/axios/generate-api.ts:38]: destructure > separate parameters

File: client/App/Exterior/Login/LoginForm/Form.tsx [hunk 3]
```diff
@@ -17,7 +17,11 @@ const Header = styled(Typography)`
 
 const initialValues = { email: '', password: '' };
 
-export const LoginForm = ({ handleSubmit }) =>
+type Props = {
```
Comment (benben12341) [inline client/App/Exterior/Login/LoginForm/Form.tsx]: I would rename the 'Props' to the name of the components as prefix. e.g 'LoginFormProps'

File: client/App/Exterior/Login/LoginForm/Form.tsx [hunk 4]
```diff
@@ -39,4 +43,4 @@ export const LoginForm = ({ handleSubmit }) =>
         </FullForm>
       </Column>
     </Formik>
-  );
+  );
```
Comment (benben12341) [inline client/App/Exterior/Login/LoginForm/Form.tsx:46]: new line at end of file ( git lab get annoyed)

File: client/App/Providers/AlertProvider/index.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+const AlertContext = createContext<AlertContext>({} as AlertContext);
+
+export const AlertProvider: FC = ({ children }) => {
+  const [message, setMessage] = useState('');
+  const [opened, setOpened] = useState(false);
+  const [severity, setSeverity] = useState<Color>('success');
+
+  const open = (message: string, severity: Color) => {
```
Comment (benben12341) [inline client/App/Providers/AlertProvider/index.tsx:17]: destructure > separate parameters

File: client/routes.tsx [hunk 6]
```diff
@@ -12,14 +11,13 @@ import {NotFound} from './App/components/NotFound';
...
-
+  // why the history was needed for?
   return (
-    <BrowserRouter history={history}>
+    <BrowserRouter>
       <App>
         <Switch>
           {routes.map((route) => (
-            <ResolvedRoute key={route.name} {...route} />
+            <ResolvedRoute key={route.name} authRequired={false} {...route} />
```
Comment (benben12341) [inline client/routes.tsx]: authRequried exist in route , its not always false

**Review by benben12341 — CHANGES_REQUESTED** [review]

File: client/App/Providers/AlertProvider/index.tsx (hunk 5 above)
Comment (OmerDital) [reply client/App/Providers/AlertProvider/index.tsx:17]: Didn't undrstand

File: client/routes.tsx (hunk 6 above)
Comment (OmerDital) [reply client/routes.tsx]: does it belong to 'shellRoutes' or to 'exteriorRoutes'?

File: client/App/utilities/axios/generate-api.ts (hunk 2 above)
Comment (OmerDital) [reply client/App/utilities/axios/generate-api.ts:38]: destructure is not needed here

**Pull request merged**